### ISSUE TITLE
Ditaa images are generated in wrong folder.

### ISSUE THREAD

**Issue by ptahchiev** [asciidoctor/asciidoctor-maven-plugin#281]
This setup:
````
                        <asciidoc>
                            <requires>
                                <require>asciidoctor-diagram</require>
                                <require>base64</require>
                                <require>time</require>
                            </requires>
                            <attributes>
                                <doctype>book</doctype>
                                <icons>font</icons>
                                <sectanchors>true</sectanchors>
                                <last-update-label>false</last-update-label>
                                <language>asciidoc</language>
                                <imagesdir>images</imagesdir>
                                <imagesoutdir>${project.build.directory}/site/images/</imagesoutdir>
                                <source-highlighter>coderay</source-highlighter>
                                <coderay-css>style</coderay-css>
                                <toclevels>2</toclevels>
                            </attributes>
                        </asciidoc>
````
of the `maven-site-plugin` fails when you work with a multi-module maven project. The `imagesoutdir` is specified as `${project.build.directory}/site/images/` and what happens is if you run you `mvn site` at the top-level because of the fact that `AsciidoctorParser` is a singleton it will render the documentation for all the modules using the same `MavenProject` - the top level maven project, and so the images will end up in the `target/` folder of the top-level project, instead of the `target/` folder of each of the modules.

After a discussion on the maven mailing list it was suggested to use:

````
	@Inject
	protected Provider<MavenProject> mavenProjectProvider;
````

to inject the maven project.

Pull-request is on it's way.

**Comment by abelsromero** [comment]
That's great! Maybe that also fixes this? https://github.com/asciidoctor/asciidoctor-maven-plugin/issues/133.
Do you have by chance a project we can use to test it? You can can drop it in the [it-test](https://github.com/asciidoctor/asciidoctor-maven-plugin/tree/master/src/it) directory and add `validate` script. Or just point me to a repo and I can include it later.

Also, can you post the link to the discussion on the maven mailing list. I'd like to have a look to learn more, and also keep it here for documentation.

**Comment by ptahchiev** [comment]
@abelsromero Here's the discussion:

http://maven.40175.n5.nabble.com/Injecting-MavenProject-td5899466.html

I'll make sure I include a test in my PR too.

**Comment by abelsromero** [comment]
Thanks :+1:

By the way, I see you're working on a previous branch and rolled back the previous changes related to https://github.com/asciidoctor/asciidoctor-maven-plugin/pull/192.
We should definitely go back to those ideas if you want in another PR. I recall Dan made some comments about the use of hyphen vs underscore but other than that, we all agreed.

I am thinking we can do a 1.5.4 release, lo align the plugin with AsciidoctorJ 1.5.4.1.

**Comment by mojavelinux** [comment]
> I am thinking we can do a 1.5.4 release, lo align the plugin with AsciidoctorJ 1.5.4.1.

:+1: 

Though, first perhaps we can get an AsciidoctorJ 1.5.5 release since 1.5.4 is the previous release of core.

**Comment by abelsromero** [comment]
> Though, first perhaps we can get an AsciidoctorJ 1.5.5 release since 1.5.4 is the previous release of core.

My first intent was to skip 1.5.4, and make a 1.5.5 aligned with AsciidoctorJ 1.5.5. But since the last seems to depend on the separation of Asciidoctorj-pdf I though of providing a 1.5.4 just now. That way we can plan the separation and release properly.

**Comment by abelsromero** [comment]
@ptahchiev Hi, let us know if you have any problem with the test part. I'll be more than happy to help with it.

**Comment by ptahchiev** [comment]
Hi @abelsromero 

I can write the test but I can't say when I will have the time to do so. Please step in to implement it if you have time. My scenario is described in the link to the maven developer list. Also I think this will fix the #192 as you correctly mentioned.

**Comment by abelsromero** [comment]
> I can write the test but I can't say when I will have the time to do so.

No problem at all, just checking. In fact I am also very busy at the moment.
Take your time.

**Comment by ptahchiev** [comment]
Guys, I've just updated my PR with the latest changes from master. Unfortunately I'm not sure how exactly to create the test-case so I would much rather if someone else steps in helps me with this. Given the fact that my commit will probably fix #133 as well I'd like to see it merged quickly :)
